src/app/routes/getInitialData/index.js

```diff
@@ -18,7 +20,47 @@ const baseUrl = onClient()
   ? getBaseUrl(window.location.origin)
   : process.env.SIMORGH_BASE_URL;
 
-const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
+export const validateRendererEnvironment = environment => {
+  const validRendererEnvironments = ['test', 'live'];
+
+  if (
+    !(
+      environment &&
+      validRendererEnvironments.includes(environment.toLowerCase())
+    )
+  ) {
+    logger.warn(
+      `Invalid parameter value [${environment}]. Usage: renderer_env=${validRendererEnvironments.join(
+        '|',
+      )}`,
+    );
+  }
+};
+
+const getPathFromUrl = url => {
+  return url.split(/[?]/)[0];
+};
+
+const checkAndAppendParams = pathname => {
```

Comment:
I've not looked at the details of the function yet but the naming suggests a code smell https://github.com/ryanmcdermott/clean-code-javascript#functions-should-do-one-thing

Reply:
So this function takes and url and splits out the query params and validates them, if not valid then logs it and then build the url with query params back up?

Reply:
Thanks - this is very useful - reminder to keep my code clean! I have reworked it.